src/vario/PageMenuAltimeter.cpp

```diff
@@ -78,21 +78,21 @@ void AltimeterMenuPage::draw() {
 }
 
 
-void AltimeterMenuPage::setting_change(int8_t dir, uint8_t state, uint8_t count) {  
+void AltimeterMenuPage::setting_change(buttons button, button_states state, uint8_t count) {  
   switch (cursor_position) {
 		case cursor_altimeter_trackGPSAlt:
       if (state == RELEASED) settings_toggleBoolNeutral(&ALT_SYNC_GPS);
       break;
     case cursor_altimeter_adjust:
-      if (dir == 0 && count == 1 && state == HELD) {  // if center button held for 1 'action time'
+      if (button == NONE && count == 1 && state == HELD) {  // if center button held for 1 'action time'
         if (settings_matchGPSAlt()) { // successful adjustment of altimeter setting to match GPS altitude
           speaker_playSound(fx_enter);  
         } else {                      // unsuccessful 
           speaker_playSound(fx_cancel);
         }
-      } else if (dir != 0) {
+      } else if (button != NONE) {
         if (state == PRESSED || state == HELD || state == HELD_LONG) {
-          baro_adjustAltSetting(dir, count);
+          baro_adjustAltSetting(button, count);
```

Comment:
We're passing an enum here but the function is actually expecting an int whose numeric value is important (e.g., negative or positive, and also how big) -- I think we actually want to call this function using that numeric value rather than casting the enum (since we don't want to rely on the enum values to have specific numeric values):

`void baro_adjustAltSetting(int8_t dir, uint8_t count)`

Reply:
It looks like we're still passing an enum here when the function is actually expecting a numeric value, and now that we're not explicitly assigning numeric values to the enum members, I think this breaks the system.  I think we should just call this function using an appropriate numeric value based on the button being pressed.  Of course this is more code/text, but I think this is the appropriate place to translate button into UI action.  One could imagine, for instance, choosing dir = 5 when state == HELD in the future (rather than dir = 1).

```suggestion
          int8_t dir = 0;
          if (button == LEFT) {
            dir = -1;
          } else if (button == RIGHT) {
            dir = 1;
          }
          baro_adjustAltSetting(dir, count);
```